python/tank_vendor/shotgun_authentication/defaults_manager.py

```diff
@@ -13,9 +13,19 @@
 
 class DefaultsManager(object):
     """
-    This class allows the ShotgunAuthenticator to get some default values when
-    authenticating a user. Having a default host and user allows having a single
-    sign-on mechanism for all Toolkit applications.
+    The defaults manager allows a client of the Shotgun Authenticator class
+    to customize various behaviors around data storage and settings.
+    
+    By default, when you construct a ShotgunAuthenticator object, it will be
+    instantiated with a standard defaults manager implementation. This will 
+    work for most cases - user session credentials will be stored in a file
+    on disk and the system maintains a concept of a current user and a current 
+    host.
+    
+    If, however, you want to implement a custom behavior around how users
+    are managed and how session credentials are stored and maintained, simply
```

Comment:
Not entirely true. The session_cache is invoked directly by the UserImpl classes, so the only thing that can be overriden is the default host, default user default login and http proxy.